client/src/components/layout/services/Fetcher.js

```diff
@@ -0,0 +1,101 @@
+class Fetcher {
+  /**
+   *
+   * @param {string} route The domain relative url of the post route
+   * @param {object} body The date object to be stringified in the fetch request
+   * @param {object=} options Options for configuring the post request:
+   * - `validationStatus`: the validation errors status code, default 422
+   * - `validationErrorParser`: parsing function used on `validationErrors' - defaults to parse "objection" validation errors
+   * @returns A response object with properties of :
+   * - `ok`: true if successfully inserted
+   * - `data`: the post response body
+   * - `validationErrors`: the parsed json validation errors
+   */
+  static async post(route, body, options = {}) {
+
+    const validationStatus = options.validationStatus ?? 422
+    const validationErrorParser = options.validationErrorParser ??translateServerErrors
+
+    const response = {
+      ok: false,
+      data: null,
+      validationErrors: null,
+    }
+
+    try {
+      const fetchOptions = {
+        method: "post",
+        headers: new Headers({ "Content-Type": "application/json" }),
+        body: JSON.stringify(body),
+      }
+      const fetchResponse = await fetch(route, fetchOptions)
+
+      if (!fetchResponse.ok) {
+        if (fetchResponse.status === validationStatus) {
+          const body = await fetchResponse.json()
+          response.validationErrors = validationErrorParser(body.errors)
+        }
+        else {
+          const errorMessage = `${fetchResponse.status} (${fetchResponse.statusText})`
+          throw Error(errorMessage)
+        }
+      } else {
+        response.ok = true
+        response.data = await fetchResponse.json()
+      }
+      return response
+    } catch (err) {
+      console.error(`Error in fetch: ${err.message}`)
+    }
+  }
+  /**
+   *
+   * @param {string} route The domain relative url of the post route
+   * @returns A response object with properties of :
+   * - `ok`: true if successfully inserted
+   * - `data`: the post response body
+   */
+  static async get(route) {
+
+    const response = {
+      ok: false,
+      data: null,
+    }
+
+    try {
+      const fetchResponse = await fetch(route)
+
+      if (fetchResponse.ok) {
+        response.ok = true
+        response.data = await fetchResponse.json()
+      } else {
+        throw Error(`${fetchResponse.status} (${fetchResponse.statusText})`)
+      }
+
+      return response
+
+    } catch (err) {
+      console.error(`Error in fetch: ${err.message}`)
+    }
+  }
+
+  // Alt version of error parser that doesn't mutate key case - may then use for <input> specific errors rather than have all in callout in one block
+  static parseObjectionValidationErrors(errors) {
+    let serializedErrors = {}
+    for (const key of Object.keys(errors)) {
+      errors[key].forEach((error) => {
+        serializedErrors[key] = `${_.startCase(key)} ${error.message}`
+      })
+    }
+    return serializedErrors
+  }
+}
+
+export default Fetcher
```

Comment:
I will allow this POJO. 

This may work for some requests, but would likely be generally a bit overkill for a project of this scope. In larger apps, your `get` and `post` requests will start to be different enough that we wouldnt make a `get` and `post` method, but instead `getReviews` and `postReview` method all defined on a `ReviewFetcher` class. 

That is to say:  if you want to DRY up your fetch logic, *first* abstract your individual fetch requests such that they are outside of your components. Then you consider abstractions for whole HTTP verbs

But coming back to this PR: this should work well, and it's a lovely implementation. If you all as a group decide you are okay with this pattern and that you each understand it, go for it. If however it's confusing to anyone, then you should have a discussion.

Reply:
if you go ahead with this, you should be using the Fetcher class for all Fetch requests. 

I would also recommend calling it `FetchClient`, but Fetcher works too.

Reply:
OK - we discussed with the group - they are OK using.  We may need to refactor slightly on the next round of commits once the branches in flight are submitted.